src/Sylius/Bundle/CoreBundle/Form/Type/Checkout/AddressType.php

```diff
@@ -29,32 +30,61 @@
 
 final class AddressType extends AbstractResourceType
 {
+    /** @var AddressComparatorInterface|null */
+    private $addressComparator;
+
+    public function __construct(string $dataClass, array $validationGroups = [], ?AddressComparatorInterface $addressComparator = null)
+    {
+        parent::__construct($dataClass, $validationGroups);
+
+        if (null === $addressComparator) {
+            @trigger_error(
+                sprintf(
+                    'Not passing an $addressComparator to %s constructor is deprecated since Sylius 1.7 and will be removed in Sylius 2.0.',
+                    __CLASS__,
+                ),
+                \E_USER_DEPRECATED
+            );
+        }
+
+        $this->addressComparator = $addressComparator;
+    }
+
     /**
      * {@inheritdoc}
      */
     public function buildForm(FormBuilderInterface $builder, array $options): void
     {
         $builder
-            ->add('differentBillingAddress', CheckboxType::class, [
-                'mapped' => false,
-                'required' => false,
-                'label' => 'sylius.form.checkout.addressing.different_billing_address',
-            ])
-            ->add('differentShippingAddress', CheckboxType::class, [
-                'mapped' => false,
-                'required' => false,
-                'label' => 'sylius.form.checkout.addressing.different_shipping_address',
-            ])
-            ->addEventListener(FormEvents::PRE_SET_DATA, static function (FormEvent $event): void {
+            ->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event): void {
                 $form = $event->getForm();
                 $order = $event->getData();
-
```

Comment:
To be reverted.

Reply:
There will be a CS check error if I do that.

Reply:
```
src/Sylius/Bundle/CoreBundle/Form/Type/Checkout/AddressType.php:63
 ----------------------------------------------------------------------------------------------------------------
 Missing variable $order before or after the documentation comment.

(SlevomatCodingStandard\Sniffs\Commenting\InlineDocCommentDeclarationSniff.MissingVariable)
```

Reply:
I moved it before the `$order` is set.

Reply:
Now I get psalm errors https://psalm.dev/docs/running_psalm/issues/RedundantConditionGivenDocblockType/
The only way I found to not get any is to remove the docblock.
Is there another way?

Reply:
I guess I'm OK with CS check error, we should disable that sniff anyway, it's more important to get Psalm passing.

Reply:
So moving this docblock just before `Assert` would be the best solution here.

Reply:
Still get that psalm error.

Reply:
The docblock is here for intelligent code completion and the assert if for the actual type checking.
The only way I can think of to achieve both is this:
```
                if (!($order instanceof OrderInterface)) {
                    // or just return;
                    throw new \InvalidArgumentException(
                        'Expected an instance of %2$s. Got: %s',
                        OrderInterface::class,
                        get_class($order)
                    );
                }
```